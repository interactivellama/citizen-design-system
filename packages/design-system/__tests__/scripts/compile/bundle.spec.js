// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/* eslint-env jest */

const fakeWebpack = config =>
  ({
    run: (config, outputPath) =>
      ({config, outputPath})
  })

const Bundle = require('../../../scripts/compile/bundle')(fakeWebpack)

describe('scripts/compile/bundle', () => {
  it('creates the right config for umd', () => {
    expect(Bundle.configs.umd.toJS()).toMatchSnapshot()
  })
  it('creates the right config for commonJS', () => {
    expect(Bundle.configs.commonJS.toJS()).toMatchSnapshot()
  })
  xit('creates the right config for chunked', done => {
    Bundle.configs.chunked
    .fork(
      e => { throw e },
      cfg => {
        expect(cfg.toJS()).toMatchSnapshot()
        done()
      }
    )
  })
  xit('compiles with webpack', done => {
    Bundle.compile(Bundle.configs.umd, '/testPath')
    .fork(
      e => { throw e },
      response => {
        // fakeWebpack just gives us our args back, but proves it "compiled" through Bundle
        expect(response.config.getIn(['output', 'libraryTarget'])).toEqual('umd')
        expect(response.outputPath).toEqual('/testPath')
        done()
      }
    )
  })
})
