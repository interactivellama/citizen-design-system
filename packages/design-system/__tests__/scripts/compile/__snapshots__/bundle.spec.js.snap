// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scripts/compile/bundle creates the right config for chunked 1`] = `
Object {
  "entry": Object {
    "components/accordion/base": "./ui/components/accordion/base/example.jsx",
    "components/activity-timeline/base": "./ui/components/activity-timeline/base/example.jsx",
    "components/activity-timeline/call": "./ui/components/activity-timeline/call/example.jsx",
    "components/activity-timeline/email": "./ui/components/activity-timeline/email/example.jsx",
    "components/activity-timeline/event": "./ui/components/activity-timeline/event/example.jsx",
    "components/activity-timeline/task": "./ui/components/activity-timeline/task/example.jsx",
    "components/alert/base": "./ui/components/alert/base/example.jsx",
    "components/app-launcher/base": "./ui/components/app-launcher/base/example.jsx",
    "components/avatar/base": "./ui/components/avatar/base/example.jsx",
    "components/badges/base": "./ui/components/badges/base/example.jsx",
    "components/breadcrumbs/base": "./ui/components/breadcrumbs/base/example.jsx",
    "components/button-groups/base": "./ui/components/button-groups/base/example.jsx",
    "components/button-icons/base": "./ui/components/button-icons/base/example.jsx",
    "components/button-icons/icon-with-dropdown": "./ui/components/button-icons/icon-with-dropdown/example.jsx",
    "components/button-icons/icon-with-hint-hover": "./ui/components/button-icons/icon-with-hint-hover/example.jsx",
    "components/button-icons/stateful": "./ui/components/button-icons/stateful/example.jsx",
    "components/buttons/base": "./ui/components/buttons/base/example.jsx",
    "components/buttons/stateful": "./ui/components/buttons/stateful/example.jsx",
    "components/buttons/with-icon": "./ui/components/buttons/with-icon/example.jsx",
    "components/cards/base": "./ui/components/cards/base/example.jsx",
    "components/checkbox-button-group/base": "./ui/components/checkbox-button-group/base/example.jsx",
    "components/checkbox-button/base": "./ui/components/checkbox-button/base/example.jsx",
    "components/checkbox-toggle/base": "./ui/components/checkbox-toggle/base/example.jsx",
    "components/checkbox/base": "./ui/components/checkbox/base/example.jsx",
    "components/combobox/base": "./ui/components/combobox/base/example.jsx",
    "components/combobox/inline-listbox": "./ui/components/combobox/inline-listbox/example.jsx",
    "components/combobox/multi-entity": "./ui/components/combobox/multi-entity/example.jsx",
    "components/combobox/readonly": "./ui/components/combobox/readonly/example.jsx",
    "components/data-tables/advanced": "./ui/components/data-tables/advanced/example.jsx",
    "components/data-tables/base": "./ui/components/data-tables/base/example.jsx",
    "components/data-tables/inline-edit": "./ui/components/data-tables/inline-edit/example.jsx",
    "components/datepickers/base": "./ui/components/datepickers/base/example.jsx",
    "components/docked-composer/base": "./ui/components/docked-composer/base/example.jsx",
    "components/docked-form-footer/base": "./ui/components/docked-form-footer/base/example.jsx",
    "components/docked-utility-bar/base": "./ui/components/docked-utility-bar/base/example.jsx",
    "components/dueling-picklist/base": "./ui/components/dueling-picklist/base/example.jsx",
    "components/dynamic-icons/ellie": "./ui/components/dynamic-icons/ellie/example.jsx",
    "components/dynamic-icons/eq": "./ui/components/dynamic-icons/eq/example.jsx",
    "components/dynamic-icons/score": "./ui/components/dynamic-icons/score/example.jsx",
    "components/dynamic-icons/strength": "./ui/components/dynamic-icons/strength/example.jsx",
    "components/dynamic-icons/trend": "./ui/components/dynamic-icons/trend/example.jsx",
    "components/dynamic-icons/waffle": "./ui/components/dynamic-icons/waffle/example.jsx",
    "components/expandable-section/base": "./ui/components/expandable-section/base/example.jsx",
    "components/feeds/base": "./ui/components/feeds/base/example.jsx",
    "components/feeds/post": "./ui/components/feeds/post/example.jsx",
    "components/file-selector/base": "./ui/components/file-selector/base/example.jsx",
    "components/file-selector/image": "./ui/components/file-selector/image/example.jsx",
    "components/file-selector/integrated": "./ui/components/file-selector/integrated/example.jsx",
    "components/files/base": "./ui/components/files/base/example.jsx",
    "components/form-element/base": "./ui/components/form-element/base/example.jsx",
    "components/form-layout/base": "./ui/components/form-layout/base/example.jsx",
    "components/form-layout/compound": "./ui/components/form-layout/compound/example.jsx",
    "components/global-header/base": "./ui/components/global-header/base/example.jsx",
    "components/global-header/notifications": "./ui/components/global-header/notifications/example.jsx",
    "components/global-navigation/navigation-bar": "./ui/components/global-navigation/navigation-bar/example.jsx",
    "components/global-navigation/navigation-tab-bar": "./ui/components/global-navigation/navigation-tab-bar/example.jsx",
    "components/icons/action": "./ui/components/icons/action/example.jsx",
    "components/icons/base": "./ui/components/icons/base/example.jsx",
    "components/icons/custom": "./ui/components/icons/custom/example.jsx",
    "components/icons/doctype": "./ui/components/icons/doctype/example.jsx",
    "components/icons/standard": "./ui/components/icons/standard/example.jsx",
    "components/input/base": "./ui/components/input/base/example.jsx",
    "components/list-builder/base": "./ui/components/list-builder/base/example.jsx",
    "components/lookups/base": "./ui/components/lookups/base/example.jsx",
    "components/lookups/multi-entity": "./ui/components/lookups/multi-entity/example.jsx",
    "components/menus/dropdown": "./ui/components/menus/dropdown/example.jsx",
    "components/modals/base": "./ui/components/modals/base/example.jsx",
    "components/page-headers/base": "./ui/components/page-headers/base/example.jsx",
    "components/page-headers/object-home": "./ui/components/page-headers/object-home/example.jsx",
    "components/page-headers/record-home": "./ui/components/page-headers/record-home/example.jsx",
    "components/page-headers/record-home-vertical": "./ui/components/page-headers/record-home-vertical/example.jsx",
    "components/panels/detail": "./ui/components/panels/detail/example.jsx",
    "components/panels/filtering": "./ui/components/panels/filtering/example.jsx",
    "components/path/base": "./ui/components/path/base/example.jsx",
    "components/picklist/base": "./ui/components/picklist/base/example.jsx",
    "components/pills/base": "./ui/components/pills/base/example.jsx",
    "components/pills/listbox-of-pill-options": "./ui/components/pills/listbox-of-pill-options/example.jsx",
    "components/popovers/base": "./ui/components/popovers/base/example.jsx",
    "components/popovers/panels": "./ui/components/popovers/panels/example.jsx",
    "components/popovers/walkthrough": "./ui/components/popovers/walkthrough/example.jsx",
    "components/progress-bar/base": "./ui/components/progress-bar/base/example.jsx",
    "components/progress-indicator/base": "./ui/components/progress-indicator/base/example.jsx",
    "components/prompt/base": "./ui/components/prompt/base/example.jsx",
    "components/publishers/base": "./ui/components/publishers/base/example.jsx",
    "components/publishers/comment": "./ui/components/publishers/comment/example.jsx",
    "components/radio-button-group/base": "./ui/components/radio-button-group/base/example.jsx",
    "components/radio-group/base": "./ui/components/radio-group/base/example.jsx",
    "components/rich-text-editor/base": "./ui/components/rich-text-editor/base/example.jsx",
    "components/scoped-tabs/base": "./ui/components/scoped-tabs/base/example.jsx",
    "components/select/base": "./ui/components/select/base/example.jsx",
    "components/spinners/base": "./ui/components/spinners/base/example.jsx",
    "components/split-view/base": "./ui/components/split-view/base/example.jsx",
    "components/tabs/base": "./ui/components/tabs/base/example.jsx",
    "components/textarea/base": "./ui/components/textarea/base/example.jsx",
    "components/tiles/base": "./ui/components/tiles/base/example.jsx",
    "components/toast/base": "./ui/components/toast/base/example.jsx",
    "components/tooltips/base": "./ui/components/tooltips/base/example.jsx",
    "components/trees/base": "./ui/components/trees/base/example.jsx",
    "components/trees/grid": "./ui/components/trees/grid/example.jsx",
    "components/vertical-navigation/base": "./ui/components/vertical-navigation/base/example.jsx",
    "components/visual-picker/coverable-content": "./ui/components/visual-picker/coverable-content/example.jsx",
    "components/visual-picker/non-coverable-content": "./ui/components/visual-picker/non-coverable-content/example.jsx",
    "components/walkthrough/header": "./ui/components/walkthrough/header/example.jsx",
    "utilities/alignment": "./ui/utilities/alignment/example.jsx",
    "utilities/borders": "./ui/utilities/borders/example.jsx",
    "utilities/box": "./ui/utilities/box/example.jsx",
    "utilities/description-list": "./ui/utilities/description-list/example.jsx",
    "utilities/floats": "./ui/utilities/floats/example.jsx",
    "utilities/grid": "./ui/utilities/grid/example.jsx",
    "utilities/horizontal-list": "./ui/utilities/horizontal-list/example.jsx",
    "utilities/hyphenation": "./ui/utilities/hyphenation/example.jsx",
    "utilities/interactions": "./ui/utilities/interactions/example.jsx",
    "utilities/margin": "./ui/utilities/margin/example.jsx",
    "utilities/media-objects": "./ui/utilities/media-objects/example.jsx",
    "utilities/name-value-list": "./ui/utilities/name-value-list/example.jsx",
    "utilities/padding": "./ui/utilities/padding/example.jsx",
    "utilities/position": "./ui/utilities/position/example.jsx",
    "utilities/scrollable": "./ui/utilities/scrollable/example.jsx",
    "utilities/sizing": "./ui/utilities/sizing/example.jsx",
    "utilities/text": "./ui/utilities/text/example.jsx",
    "utilities/themes": "./ui/utilities/themes/example.jsx",
    "utilities/truncation": "./ui/utilities/truncation/example.jsx",
    "utilities/vertical-list": "./ui/utilities/vertical-list/example.jsx",
    "utilities/visibility": "./ui/utilities/visibility/example.jsx",
  },
  "externals": Object {
    "lodash": "lodash",
    "react": "React",
    "react-dom": "ReactDOM",
  },
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "loader": "babel-loader",
        "options": Object {
          "babelrc": false,
          "plugins": Array [
            "styled-jsx/babel",
            "transform-object-rest-spread",
          ],
          "presets": Array [
            "es2015",
            "react",
          ],
        },
        "test": /\\\\\\.jsx\\?\\$/,
      },
    ],
  },
  "output": Object {
    "filename": "[name].js",
    "library": "SLDS",
    "publicPath": "assets/scripts/bundle/",
  },
  "plugins": Array [
    CommonsChunkPlugin {
      "async": undefined,
      "children": undefined,
      "chunkNames": Array [
        "common",
      ],
      "filenameTemplate": undefined,
      "ident": "/Users/blonsdorf/Documents/design-system-internal/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js0",
      "minChunks": 2,
      "minSize": undefined,
      "selectedChunks": undefined,
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".jsx",
    ],
  },
}
`;

exports[`scripts/compile/bundle creates the right config for commonJS 1`] = `
Object {
  "entry": "./scripts/compile/entry.js",
  "externals": Object {
    "lodash": "lodash",
    "react": "React",
    "react-dom": "ReactDOM",
  },
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "loader": "babel-loader",
        "options": Object {
          "babelrc": false,
          "plugins": Array [
            "styled-jsx/babel",
            "transform-object-rest-spread",
          ],
          "presets": Array [
            "es2015",
            "react",
          ],
        },
        "test": /\\\\\\.jsx\\?\\$/,
      },
    ],
  },
  "output": Object {
    "filename": "slds.common.js",
    "libraryTarget": "commonjs2",
    "publicPath": "assets/scripts/bundle/",
  },
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".jsx",
    ],
  },
}
`;

exports[`scripts/compile/bundle creates the right config for umd 1`] = `
Object {
  "entry": "./scripts/compile/entry.js",
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "loader": "babel-loader",
        "options": Object {
          "babelrc": false,
          "plugins": Array [
            "styled-jsx/babel",
            "transform-object-rest-spread",
          ],
          "presets": Array [
            "es2015",
            "react",
          ],
        },
        "test": /\\\\\\.jsx\\?\\$/,
      },
    ],
  },
  "output": Object {
    "filename": "slds.umd.js",
    "library": "SLDS",
    "libraryTarget": "umd",
    "publicPath": "assets/scripts/bundle/",
  },
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".jsx",
    ],
  },
}
`;
